incremental-font-transfer/src/table_keyed.rs

```diff
@@ -0,0 +1,441 @@
+        else {
+            return Err(PatchingError::InvalidPatch(
+                "Patch offsets are not in sorted order.",
+            ));
+        };
+
+        if stream_length as usize > table_patch.brotli_stream().len() {
+            return Err(PatchingError::PatchParsingFailed(ReadError::OutOfBounds));
+        }
+
+        let tag = table_patch.tag();
+        if !processed_tables.insert(tag) {
+            // Table has already been processed.
+            continue;
+        }
+
+        if table_patch.flags().contains(TablePatchFlags::DROP_TABLE) {
+            // Table will not be copied, skip any further processing.
+            continue;
+        }
+
+        let replacement = table_patch.flags().contains(TablePatchFlags::REPLACE_TABLE);
+        let new_table = apply_table_patch(font, table_patch, stream_length, replacement)?;
+        font_builder.add_raw(tag, new_table);
+    }
+
+    copy_unprocessed_tables(font, processed_tables, &mut font_builder);
+
+    Ok(font_builder.build())
+}
+
+fn apply_table_patch(
+    font: &FontRef,
+    table_patch: TablePatch,
+    stream_length: u32,
+    replacement: bool,
+) -> Result<Vec<u8>, PatchingError> {
+    let stream_length = stream_length as usize;
+    let base_data = font.table_data(table_patch.tag());
+    let stream = if table_patch.brotli_stream().len() >= stream_length {
+        &table_patch.brotli_stream()[..stream_length]
+    } else {
+        return Err(PatchingError::InvalidPatch(
+            "Brotli stream is larger then the maxUncompressedLength field.",
+        ));
+    };
+    let r = match (base_data, replacement) {
+        (Some(base_data), false) => shared_brotli_decode(
+            stream,
+            Some(base_data.as_bytes()),
+            table_patch.max_uncompressed_length() as usize,
+        ),
+        (None, false) => {
+            return Err(PatchingError::InvalidPatch(
+                "Trying to patch a base table that doesn't exist.",
+            ))
+        }
+        _ => shared_brotli_decode(stream, None, table_patch.max_uncompressed_length() as usize),
+    };
+
+    r.map_err(PatchingError::from)
+}
+
+pub(crate) fn copy_unprocessed_tables<'a>(
+    font: &FontRef<'a>,
+    processed_tables: BTreeSet<Tag>,
+    font_builder: &mut FontBuilder<'a>,
+) {
+    font.table_directory
+        .table_records()
+        .iter()
+        .map(|r| r.tag())
+        .filter(|tag| !processed_tables.contains(tag))
+        .filter_map(|tag| Some((tag, font.table_data(tag)?)))
+        .for_each(|(tag, data)| {
+            font_builder.add_raw(tag, data);
+        });
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use font_test_data::ift::{noop_table_keyed_patch, table_keyed_patch};
+    use read_fonts::FontData;
+    use read_fonts::FontRead;
+    use read_fonts::FontRef;
+    use read_fonts::ReadError;
+    use write_fonts::FontBuilder;
+
+    const IFT_TABLE: &[u8] = "IFT PATCH MAP".as_bytes();
```

Comment:
you can write a byte literal be prepending a string literal with 'b', e.g. `b"IFT PATCH MAP"`.